app/views/posts/index.html.erb

```diff
@@ -5,7 +5,7 @@
         <p>There is no post yet.</p>
     <% end %>
 
-    <% @user.posts.each do |post| %>
```

Comment:
- [ ] Good job trying to fix the `n + 1 problems`. However for Rails best practices and as required for the project, I think it would be better if the `includes` method is used in the controller to specify associations that should be eagerly loaded when retrieving records from the database. By including associations in the initial query, Rails can fetch the associated data in a more efficient way, avoiding n+1 queries. 
For Example in the `posts#index` actions you can use:
```ruby
def index
  @posts = @user.posts.includes(:comments)
end
# instead of 
def index
  @posts = @user.posts
end
```